packages/compass-instance/src/plugin.jsx

```diff
@@ -1,90 +1,19 @@
 const React = require('react');
-const filter = require('lodash.filter');
-const PropTypes = require('prop-types');
-const { TabNavBar, UnsafeComponent } = require('hadron-react-components');
-const { track } = require('@mongodb-js/compass-logging').createLoggerAndTelemetry('COMPASS-INSTANCE-UI');
-
-function trackingIdForTabName(name) {
-  return name.toLowerCase().replace(/ /g, '_');
-}
-
-/**
- * Represents the instance view.
- */
-class InstanceComponent extends React.Component {
-  static displayName = 'InstanceComponent';
-  static propTypes = {
-    isDataLake: PropTypes.bool.isRequired
-  };
-
-
-  /**
-   * Instantiate the instance component.
-   *
-   * @param {Object} props - The properties.
-   */
-  constructor(props) {
-    super(props);
-    this.state = { activeTab: 0 };
-  }
-
-  /**
-   * Handle the tab click.
-   *
-   * @param {Number} idx - The index of the clicked tab.
-   */
-  onTabClicked(idx, name) {
-    if (this.state.activeTab === idx) {
-      return;
-    }
-    global.hadronApp.appRegistry.emit('compass:screen:viewed', { screen: name });
-    track('Screen', { name: trackingIdForTabName(name) });
-    this.setState({ activeTab: idx });
-  }
-
-  /**
-   * Setup the instance level tabs.
-   */
-  setupTabs() {
-    const instanceTabs = global.hadronApp.appRegistry.getRole('Instance.Tab');
-    const roles = filter(instanceTabs, (role) => {
-      return !(this.props.isDataLake && role.name === 'Performance');
-    });
-
-    const tabs = roles.map((role) => role.name);
-    const views = roles.map((role, i) => {
-      return (
-        <UnsafeComponent component={role.component} key={i} />
-      );
-    });
-
-    this.tabs = tabs;
-    this.views = views;
-    if (tabs.length > 0) {
-      track('Screen', { name: trackingIdForTabName(tabs[0]) });
-    }
-  }
-
-  /**
-   * Render the instance component.
-   *
-   * @returns {React.Component} The rendered component.
-   */
-  render() {
-    this.setupTabs();
-    return (
-      <div className="rtss">
-        <TabNavBar
-          data-test-id="instance-tabs"
-          aria-label="Instance Tabs"
-          tabs={this.tabs}
-          views={this.views}
-          activeTabIndex={this.state.activeTab}
-          onTabClicked={this.onTabClicked.bind(this)}
-          mountAllViews={false} />
-      </div>
-    );
-  }
-}
-
-module.exports = InstanceComponent;
+const { Provider, connect } = require('react-redux');
+const store = require('./stores');
+const { InstanceComponent } = require('./components/instance');
+
+const ConnectedInstanceComponent = connect(
+  (state) => state,
+  () => ({})
+)(InstanceComponent);
+
+const InstancePlugin = () => {
+  return (
+    <Provider store={store}>
+      <ConnectedInstanceComponent />
```

Comment:
This plugin/package looks like one component and one store - would it make sense to just move it into compass-home? I think down the line if we change the performance tab to being opened from the sidebar this package would be removed as well.
Maybe out of the scope of this ticket, but I figured I'd bring it up since the diff here already is large.

Reply:
I don't think it would in the current state of the app, from my perspective instance state doesn't belong in the root state of the home plugin, there is no other pattern we have to have this state outside of home root (and still have access to appRegistry), and I don't want to introduce more different ways of state handling in the application. Currently instance plugin is similar to collection or database plugin: compass-home handles (to the best of its abilities taking into account that not all namespace / workspace changes are communicated through the event bus) active workspace and just responsible for rendering an appropriate one, then every plugin is responsible for rendering itself and handling possible errors that might occur during data fetching, this PR keeps the logic in line with how these other plugins behave.

> I think down the line if we change the performance tab to being opened from the sidebar this package would be removed as well.

I think when this happens we would need to really invest some time into normalizing how we handle workspaces / current workspace first and then maybe we make performance and databases separate plugins / components and there is no point in having this plugin here.

In general I don't think merging everything into compass-home is the best thing we can do here, for example in this PR most of the work was decoupling instance state out of compass-home because even slight changes of handling instance fetching errors required addressing cascading changes to literally all parts of compass-home even though none of them really depend on this stuff. I don't want us running into the same issues in the future, there should be some logical separation of state for different parts of the app otherwise we will just constantly cause merge conflicts while working on seemingly unrelated parts of the application. Unfortunately the best way of doing this separation right now is our plugin system.

Reply:
Makes sense to me. Definitely worth planning it out when we look into those potential changes and normalizing how the workspaces work.